src/lib/support/Span.h

```diff
@@ -265,10 +274,19 @@ class FixedSpan
         static_assert(M >= N, "Passed-in buffer too small for FixedSpan");
     }
 
+    // Creates a (potentially mutable) FixedSpan view of an std::array
     template <class U, typename = std::enable_if_t<sizeof(U) == sizeof(T) && std::is_convertible<U *, T *>::value>>
     constexpr FixedSpan(std::array<U, N> & arr) : mDataBuf(arr.data())
     {}
 
+    template <class U, typename = std::enable_if_t<sizeof(U) == sizeof(T) && std::is_convertible<U *, T *>::value>>
```

Comment:
nit: You might have wanted to keep the code consistent, but we already use C++17, so you could use `std::is_convertible_v` in the new code as well.

Reply:
Yeah I thought I'd keep it the same within that file at least. Maybe we should do a general clean up at some point to bulk-replace them.